src/components/header/share-modal/renderer.tsx

```diff
@@ -25,8 +27,15 @@ class ShareModal extends React.PureComponent<Props, State> {
   }
 
   public exportURL() {
+    let spec;
+    if (this.props.mode === Mode.VegaLite) {
+      spec = this.props.vegaLiteSpec;
+    } else {
+      spec = this.props.vegaSpec;
+    }
+    spec.config = this.props.config;
     const serializedSpec =
-      LZString.compressToEncodedURIComponent(this.props.editorString) + (this.state.fullScreen ? '/view' : '');
+      LZString.compressToEncodedURIComponent(stringify(spec)) + (this.state.fullScreen ? '/view' : '');
```

Comment:
@siddhant1 I suggest using template literals.

Reply:
This is actually not what we want. We do want to save the spec like it was formatted. 

If you really feel like changing this, then we should store the spec compressed without whitespace.